### PR TITLE
Issue #998: Migrate console commands from sample-project repo

### PR DESCRIPTION
**Pull request by Vinai**
Most done. A few more, then I'm done.

### CODE REVIEW

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace LizardsAndPumpkins\ConsoleCommand\Command\EventProcessingTimeAverage;
+
+class LogfileReader
+{
+
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php]: Superfluous newline.

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public function getEventHandlerProcessingTimes(string $logFilePath): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php]: Missing PHPDoc.

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        $eventHandlers = [];
+        foreach ($this->readDomainEventHandlerProcessingTimes($logFilePath) as $record) {
+            list($domainEventHandler, $time) = $record;
+            $eventHandlers[$domainEventHandler][] = $time;
+        }
+
+        return $eventHandlers;
+    }
+
+    private function readDomainEventHandlerProcessingTimes($logFilePath)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php]: Missing either type hint or PHPDoc.

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    public function getEventHandlerProcessingTimes(string $logFilePath): array
+    {
+        $eventHandlers = [];
+        foreach ($this->readDomainEventHandlerProcessingTimes($logFilePath) as $record) {
+            list($domainEventHandler, $time) = $record;
+            $eventHandlers[$domainEventHandler][] = $time;
+        }
+
+        return $eventHandlers;
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php]: I would prefer to see an `array_reduce` here but I can see why you use `foreach` here.

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/ProcessingTimeTableDataBuilder.php [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        'Total Sec'   => 'total',
+        'Average Sec' => 'avg',
+    ];
+    
+    private static $sortDirections = [
+        -1 => 'asc',
+        1 => 'desc'
+    ];
+
+    public function buildSortedTableData(array $eventHandlerProcessingTimes, string $sortBy, string $direction): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/ProcessingTimeTableDataBuilder.php]: Missing PHPDoc.

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/ProcessingTimeTableDataBuilder.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    /**
+     * @param array[] $eventHandlerProcessingTimes
+     * @return array[]
+     */
+    private function buildTableData(array $eventHandlerProcessingTimes): array
+    {
+        return array_map(
+            function ($handler) use ($eventHandlerProcessingTimes) {
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/ProcessingTimeTableDataBuilder.php]: Wouldn't it fit into a previous line?

File: src/ConsoleCommand/Command/ImportContentBlockConsoleCommand.php [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+                'noValue'     => true,
+            ],
+            'importDirectory' => [
+                'description' => 'Path to directory with *.html files to import',
+                'required'    => true,
+            ],
+        ]);
+    }
+
+    protected function execute(CLImate $CLImate)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ImportContentBlockConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ImportContentBlockConsoleCommand.php [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        $this->factory = $factory;
+        $this->setCLImate($CLImate);
+    }
+
+    /**
+     * @param CLImate $climate
+     * @return array[]
+     */
+    protected function getCommandLineArgumentsArray(CLImate $climate): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ImportContentBlockConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ImportContentBlockConsoleCommand.php [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+        every($contentFileNames, function (string $contentFileName) {
+            $blockId = $this->createContentBlockIdBasedOnFileName($contentFileName);
+            $blockContent = file_get_contents($contentFileName);
+            $keyGeneratorParams = $this->createKeyGeneratorParamsBasedOnFileName($contentFileName);
+            
+            $this->addCommandsForEachContext($blockId, $blockContent, $keyGeneratorParams);
+        });
+    }
+
+    private function addCommandsForEachContext(ContentBlockId $blockId, string $blockContent, array $keyGeneratorParams)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ImportContentBlockConsoleCommand.php]: Missing PHPDoc.

File: src/ConsoleCommand/Command/ReportEventProcessingTimeAverageConsoleCommand.php [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        $this->setCLImate($CLImate);
+        $this->processingTimeTableDataBuilder = $masterFactory->createProcessingTimeTableDataBuilder();
+        $this->logFileReader = $masterFactory->createDomainEventProcessingTimesLogFileReader();
+    }
+
+    /**
+     * @param CLImate $climate
+     * @return array[]
+     */
+    protected function getCommandLineArgumentsArray(CLImate $climate): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ReportEventProcessingTimeAverageConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ReportEventProcessingTimeAverageConsoleCommand.php [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+                ],
+                'logfile'   => [
+                    'description' => 'Log file',
+                    'required'    => true,
+                ],
+            ]
+        );
+    }
+
+    protected function execute(CLImate $climate)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ReportEventProcessingTimeAverageConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ReportQueueCountConsoleCommand.php [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    private $masterFactory;
+
+    public function __construct(MasterFactory $factory, CLImate $climate)
+    {
+        $this->masterFactory = $factory;
+        $this->setCLImate($climate);
+    }
+
+    protected function execute(CLImate $climate)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ReportQueueCountConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ReportUrlKeysConsoleCommand.php [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+    {
+        $this->factory = $factory;
+        $this->setCLImate($CLImate);
+    }
+
+    /**
+     * @param CLImate $CLImate
+     * @return array[]
+     */
+    protected function getCommandLineArgumentsArray(CLImate $CLImate): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ReportUrlKeysConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ReportUrlKeysConsoleCommand.php [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+                'dataVersion' => [
+                    'description'  => 'List url keys for the given catalog data version',
+                    'defaultValue' => self::CURRENT_VERSION,
+                    'required'     => false,
+                ],
+            ]
+        );
+    }
+
+    protected function execute(CLImate $climate)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ReportUrlKeysConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ReportUrlKeysConsoleCommand.php [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+    {
+        $version = $this->getDataVersionToDisplay();
+        $urlKeys = $this->getUrlKeys($version, $this->getArg('type'));
+        $formattedUrlKeys = $this->formatUrlKeys($urlKeys);
+        
+        $this->outputMessage('URL keys:');
+        $this->outputArray($formattedUrlKeys);
+    }
+
+    private function getUrlKeys(string $version, string $type): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ReportUrlKeysConsoleCommand.php]: Missing DocBlock?

File: src/ConsoleCommand/Command/ShutdownConsumerConsoleCommand.php [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        $this->factory = $factory;
+        $this->setCLImate($climate);
+    }
+    
+    /**
+     * @param CLImate $climate
+     * @return array[]
+     */
+    protected function getCommandLineArgumentsArray(CLImate $climate): array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ShutdownConsumerConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/ShutdownConsumerConsoleCommand.php [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+                ],
+                'pid'  => [
+                    'description'  => 'numeric PID',
+                    'required'     => true,
+                ],
+            ]
+        );
+    }
+
+    protected function execute(CLImate $climate)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/ShutdownConsumerConsoleCommand.php]: Maybe final?

File: src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php (hunk 4 above)
Comment (Vinai) [inline src/ConsoleCommand/Command/EventProcessingTimeAverage/LogfileReader.php]: Yeah, I don't want the complete log file to be held in memory.

**Comment by Vinai** [comment]
Ready for next review.

**Pull request merged**